app/models/grda_warehouse/client_file.rb

```diff
@@ -39,30 +39,47 @@ class ClientFile < GrdaWarehouse::File
 
     scope :visible_by?, ->(user) do
       can_view_homeless_verification_pdfs = GrdaWarehouse::Config.get(:verified_homeless_history_visible_to_all) || user.can_generate_homeless_verification_pdfs?
+      is_own_file = arel_table[:user_id].eq(user.id)
+      is_verified_homeless_history = arel_table[:id].in(Arel.sql(verified_homeless_history.select(:id).to_sql))
+      is_not_verified_homeless_history = arel_table[:id].not_in(Arel.sql(verified_homeless_history.select(:id).to_sql))
+      is_consent_form = arel_table[:id].in(Arel.sql(consent_forms.select(:id).to_sql))
+      has_full_housing_release = arel_table[:client_id].in(
+        Arel.sql(GrdaWarehouse::Hud::Client.full_housing_release_on_file.select(:id).to_sql),
+      )
+
       # If you can see all client files, show everything
       if user.can_manage_client_files?
         current_scope
       # If all you can see are window files:
       #   show those with full releases and those you uploaded
       elsif user.can_manage_window_client_files? || user.can_use_separated_consent?
-        sql = arel_table[:client_id].in(
-          Arel.sql(GrdaWarehouse::Hud::Client.full_housing_release_on_file.select(:id).to_sql),
-        ).or(arel_table[:user_id].eq(user.id))
-
-        sql = sql.or(arel_table[:id].in(Arel.sql(consent_forms.select(:id).to_sql))) if GrdaWarehouse::Config.get(:consent_visible_to_all)
-        sql = sql.or(arel_table[:id].in(Arel.sql(verified_homeless_history.select(:id).to_sql))) if can_view_homeless_verification_pdfs
+        sql = is_own_file
+        sql = sql.or(has_full_housing_release.and(is_not_verified_homeless_history))
```

Comment:
This is probably really inefficient, but I wanted to be clear about the logic changed in https://github.com/greenriver/hmis-warehouse/pull/2355/commits/1e024bbd13d5dc574f8d5ee3f543bd167cc2df80

Basically we don't want to blanket include all files if the client has a full housing release, because we limit it further down for homeless history files specifically (when config is set to `:release`). @eanders

Reply:
I think that looks correct.